mm.c

```diff
@@ -161,11 +161,42 @@ static void *extend_heap(size_t words) {
 	enough to hold the amount of words specified.
 	Returns the pointer to that block.
 	Does not take the block out of the free list.
+	Does not extend heap.
 	Returns the pointer to the block.
 	Returns NULL if block large enough is not found.
 */
 static void *find_free_block(size_t words) {
-
+	char **bp;
+	size_t index = find_free_list_index(words);
+
+	// check if the current index has any blocks big enough
+	if ((bp = GET_FREE_LIST_PTR(index)) != NULL && GET_SIZE(bp) >= words) {
+		// iterate until smallest element in this index
+		while (GET_SUCC(bp) != NULL && GET_SIZE(GET_SUCC(bp)) >= words) {
```

Comment:
is it ok to rewrite this loop to return the first free block whose size is equal "words" ? As of now, it will find the last such block, which is slower than returning the first such block.